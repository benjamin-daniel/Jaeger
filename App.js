import React from 'react';
import {Text} from 'react-native';
import {ThemeProvider} from '@shopify/restyle';

import theme from 'constants/theme';

const App = () => {
  return (
    <>
      <ThemeProvider theme={theme}>
        <Text>Omo</Text>
      </ThemeProvider>
    </>
  );
};

export default App;
