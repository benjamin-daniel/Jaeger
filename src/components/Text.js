import React, {ReactNode} from 'react';
import {createText} from '@shopify/restyle';
import {Text as RNText, TextProps as RNTextProps} from 'react-native';

const _Text = (props) => <RNText {...props} />;

export const Text = createText(_Text);

Text.defaultProps = {
  variant: 'bodyText',
  // color: 'bodyText',
  // fontFamily: 'Telex-Regular',
};
