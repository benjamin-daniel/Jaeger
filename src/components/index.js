export * from './Box';
export * from './ErrorText';
export * from './Input';
