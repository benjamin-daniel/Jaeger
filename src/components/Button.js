import React from 'react';
import {
  TouchableOpacity,
  ActivityIndicator,
  StyleSheet,
  Platform,
  TextStyle,
  ViewStyle,
} from 'react-native';
import {useTheme} from '@shopify/restyle';

import {Text} from './Text';
import {Box} from './Box';
import {palette} from 'constants/theme';

export const Button = ({
  disabled,
  loading,
  variant = 'primary',
  text,
  onPress,
  color: buttonColorName = palette.white,
  textVariant = 'button',
}) => {
  const theme = useTheme();
  const variantProps = theme.buttonVariants[variant];
  const disabledProps = disabled ? variantProps.disabled || {} : {};
  const themedStyles = {...variantProps, ...disabledProps};
  const {
    fontSize,
    fontWeight,
    fontFamily,
    color,
    borderWidth,
    height,
  } = themedStyles;
  const textColor = themedStyles.textColor;
  const buttonColor = buttonColorName && theme.colors[buttonColorName];

  const onPressHandler = loading ? () => {} : onPress;
  const content = (
    <Box
      borderRadius={5}
      alignItems="center"
      justifyContent="center"
      style={{
        backgroundColor: buttonColor || color,
        minHeight: height,
        borderWidth,
        borderColor: buttonColor,
      }}
      paddingHorizontal="m"
      flexDirection="row">
      {loading ? (
        <ActivityIndicator color={textColor} />
      ) : (
        <>
          <Text
            variant={textVariant}
            style={{
              color: textColor || buttonColor,
              fontWeight,
              fontFamily,
              fontSize,
            }}>
            {text}
          </Text>
        </>
      )}
    </Box>
  );
  return (
    <TouchableOpacity
      onPress={onPressHandler}
      style={[styles.stretch]}
      disabled={loading}
      {...{disabled}}>
      {content}
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  button: {
    paddingVertical: 16,
  },
  stretch: {
    alignSelf: 'stretch',
  },
  content: {
    textAlign: 'center',
  },
});
