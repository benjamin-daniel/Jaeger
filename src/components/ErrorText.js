import React from 'react';
import {Text} from './Text';

export const ErrorText = ({error, touched, ErrorTextStyles}) => (
  <>
    {touched && error && (
      <Text fontSize={12} color="error" style={[ErrorTextStyles]}>
        {error}
      </Text>
    )}
  </>
);
