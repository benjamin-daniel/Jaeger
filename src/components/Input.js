import React, {useState} from 'react';
import {TextInput, StyleSheet, TouchableOpacity} from 'react-native';
import PropTypes from 'prop-types';
import Feather from 'react-native-vector-icons/Feather';
import {useTheme} from '@shopify/restyle';
import {Box} from './Box';
import {ErrorText} from './ErrorText';
/**
 *
 * @see https://lefkowitz.me/visual-guide-to-react-native-textinput-keyboardtype-options/
 */

const DEFAULT_STYLE = {
  //   fontFamily: 'Telex-Regular',
};

export const Input = ({
  inputStyle,
  containerStyle,
  touched,
  error,
  passwordIcon,
  variant,
  labelVariant,
  Icon,
  innerContainerProps,
  LeftIcon,
  RightIcon,
  ErrorTextStyles,
  nospace,
  ...props
}) => {
  const [state, setState] = useState({
    icon: 'eye-off',
    password: !!Icon,
  });

  const theme = useTheme();
  const variantProps = theme.inputVariants[variant];
  const ICON_SIZE = 20;
  const realInputStyle = StyleSheet.flatten([
    DEFAULT_STYLE,
    variantProps,
    inputStyle,
  ]);
  const changeIcon = () => {
    setState((prevState) => ({
      icon: prevState.icon === 'eye' ? 'eye-off' : 'eye',
      password: !prevState.password,
    }));
  };
  return (
    <Box marginVertical={nospace ? 'none' : 's'}>
      <Box
        borderWidth={1}
        borderColor={touched && error ? 'error' : 'black'}
        flexDirection="row"
        alignItems="center"
        height={40}
        padding="m"
        backgroundColor="mainBackground"
        borderRadius={5}
        {...innerContainerProps}>
        {LeftIcon}
        <Box flex={1}>
          <TextInput
            style={realInputStyle}
            secureTextEntry={state.password}
            {...props}
          />
        </Box>
        {RightIcon}
        {Icon && (
          <TouchableOpacity onPress={() => changeIcon()}>
            <Box
              marginHorizontal="m"
              alignItems="center"
              justifyContent="center">
              <Feather size={ICON_SIZE} name={state.icon} />
            </Box>
          </TouchableOpacity>
        )}
      </Box>
      <ErrorText {...{touched, error, ErrorTextStyles}} />
    </Box>
  );
};
