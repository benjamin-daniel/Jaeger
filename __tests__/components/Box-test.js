/**
 * @format
 */

import 'react-native-gesture-handler';
import 'react-native';
import React from 'react';
import {Dimensions} from 'react-native';
import {Container} from '../../App';
import {View as RNView} from 'react-native';

import {Box} from 'components';
import renderer from 'react-test-renderer';

jest.mock('react-native/Libraries/Utilities/Dimensions', () => {
  return {
    get: () => ({
      width: 375,
      height: 667,
    }),
    addEventListener: jest.fn(),
  };
});

describe('Box component test', () => {
  beforeEach(() => {
    Dimensions.addEventListener.mockClear();
  });

  it('renders correctly', async () => {
    const tree = await renderer
      .create(
        <Container>
          <Box>Many</Box>
        </Container>,
      )
      .toJSON();
    await expect(tree).toMatchSnapshot();
  });

  it('passes styles based on the given props', async () => {
    const {root} = renderer.create(
      <Container>
        <Box variant="none" opacity={0.5}>
          hey
        </Box>
      </Container>,
    );
    await expect(root.findByType(Box).props).toEqual(
      expect.objectContaining({opacity: 0.5}),
    );
  });
});
