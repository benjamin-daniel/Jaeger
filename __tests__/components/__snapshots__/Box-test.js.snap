// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box component test renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "display": "flex",
      },
    ]
  }
>
  Many
</View>
`;
